Blizzard.IO.RabbitMQ/Implementaions/NetqRabbitConsumer.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using Blizzard.IO.Core;
+using Blizzard.IO.RabbitMQ.Entities;
+using EasyNetQ;
+using EasyNetQ.Topology;
+
+namespace Blizzard.IO.RabbitMQ.Implementaions
+{
+    public class NetqRabbitConsumer<TData> : IConsumer<TData>, IConsumerWithMetadata<TData, RabbitMessageProperties>
+    {
+        private readonly IBus _netqBus;
+        private readonly IDeserializer<TData> _deserializer;
+        private readonly IConcreteTypeDeserializer<TData> _concreteTypeDeserializer;
```

Comment:
Also I think we have a problem with this interface (it is my fault I told Yair to call it like this) this is AbstractTypeDeserializer not ConcreteTypeDeserializer Right?

Reply:
yea

Reply:
f